### ISSUE TITLE
[bug] Revisions are not updated

### ISSUE THREAD

**Issue by ghost** [conan-io/conan#9369]
Hello,

I am producing revisions for versions (enabled on server and client).
In short - no matter what I tried to do, I dont seem to be able to get new version, only if I delete packages locally and pull new.
We are using SCM for revisions, therefore the revision is GIT sha and not the default hash.

### Environment Details
  * Operating System+version: Ubuntu 20.04
  * Conan version: 1.35.1
  * Python version: 3.9.5

### Steps to reproduce
Create intial revision
pull on client via command:
conan install "pkg/12.0.0@user/stable" --update -r server -o app=app -o variant=debug -o server_build=True
Create new version
Run again conan install "pkg/12.0.0@user/stable" --update -r server -o app=app -o variant=debug -o server_build=True

Output:
Installing package: x/y@z/stable
Requirements
    x/y@z/stable from 'server' - Cache
Packages
    x/y@z/stable:23d871b02677d94407dd30d3bcf29c0dabdfa428 - Cache
Installing (downloading, building) binaries...
Already installed!

**list local revision (returns old one)**
```
root@ca8832285a96:/usr/local# conan search "application/4.5.3@user/stable" --rev
Revisions for 'application/4.5.3@user/stable':
4ea97533e216bc63ac1fd9e4d96f15e6d44342a1 (2021-08-03 16:25:31 UTC)
```

**list revisions on server - notice new one**
```
root@ca8832285a96:/usr/local# conan search "application/4.5.3@user/stable" --rev -r server
Revisions for 'application/4.5.3@user/stable' at remote 'server':
4b1ece5f4134ad22e57262a5ee68acb5a43ce65b (2021-08-03 16:39:58 UTC)
4ea97533e216bc63ac1fd9e4d96f15e6d44342a1 (2021-08-03 16:25:31 UTC)
```

**finally try to update**
```
root@ca8832285a96:/usr/local# conan install "application/4.5.3@user/stable" --update -r server -o app=application -o variant=debug -o server_build=True
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++
compiler.version=9
os=Linux
os_build=Linux
[options]
app=application
server_build=True
variant=debug
[build_requires]
[env]

Installing package: application/4.5.3@user/stable
Requirements
    application/4.5.3@user/stable from 'server' - Cache
Packages
    application/4.5.3@user/stable:23d871b02677d94407dd30d3bcf29c0dabdfa428 - Cache

Installing (downloading, building) binaries...
application/4.5.3@user/stable: Already installed!
```

**Comment by lasote** [comment]
I don't know if that is the issue but in the "finally try to update" you are using a different user/channel `application/4.5.3@user/stable` vs `application/4.5.3@d13-rf/stable`

**Comment by ghost** [comment]
No, this was just a typo because I have tried to remove our project specific names from the commands.
All commands are executed as they should be with propper package name, user and channel

**Comment by ghost** [comment]
I did a workaround - running `conan remove "application*" -f` before the install command

**Comment by lasote** [comment]
What are you using on the server-side? Artifactory or the conan_server?

**Comment by ghost** [comment]
It is JFrog Artifactory version 6.17.0

**Comment by lasote** [comment]
It might be an issue with a different time between the server and the client. Currently to know if a revision is newer, Conan reads the `export/conanmanifest.txt`. The first number in the file is the timestamp, so it is compared with the local one (the one in the conan local cache). Could you check if the timestamp (for the conanmanifest.txt of that latest revision) in the server file is older than the one in the local cache? And/or if the time is more or less synchronized?

We know that this is an issue and we have improved it for Conan 2.0, but currently, the timestamp of the manifest file is used.

**Comment by ghost** [comment]
Local conanmanifest content is:
```
1628079118
conanfile.py: 8ad26f1ee4e3dda23e902c5357d108bf
```

Server side content:
```
1628154486
conanfile.py: 8ad26f1ee4e3dda23e902c5357d108bf
```

The number on server is greater.

Also conan search command confirms this (that locally it is an older version than on server).
Local search:
```
PS C:\work\repos\repo_a> conan search app/12.0.0@user/stable --rev
Revisions for 'app/12.0.0@user/stable':
**f88edded68b36aac1ef21c76d179feb47a8e24ad (2021-08-04 14:12:00 UTC)**
```

server search:
```
PS C:\work\repos\repo_a> conan search app/12.0.0@user/stable -r virtual-remote --rev
Revisions for 'app/12.0.0@user/stable' at remote 'virtual-remote':
e2ab02645e3e47aa5d53ea38192f317b9113e310 (2021-08-05 11:08:08 UTC)
8407b891405ce58377c95efde137cc355c8b456e (2021-08-05 09:48:59 UTC)
**f88edded68b36aac1ef21c76d179feb47a8e24ad (2021-08-04 14:12:00 UTC)**
e81d1944591154e5711dd8a2ff33a09f3df405e2 (2021-08-04 13:11:34 UTC)
0ee873b6aa7ea50618c4683cce23530ecc982a26 (2021-08-04 12:02:10 UTC)
4b1ece5f4134ad22e57262a5ee68acb5a43ce65b (2021-08-03 14:19:48 UTC)
f795219c08d2adfbd1c58baf1f1daca39507a5aa (2021-08-03 12:58:53 UTC)
4ea97533e216bc63ac1fd9e4d96f15e6d44342a1 (2021-08-03 12:39:12 UTC)
fc395b4147d206157d61a7d5e01ec5063b0fe094 (2021-08-03 10:48:22 UTC)
b1807c040c4386e1d2ed28c36bb969e9e457833a (2021-08-02 13:28:26 UTC)
bc7d6f0804a59298f4f017aeb6d24528f34adc97 (2021-08-02 13:02:00 UTC)
```

The UTC time is older for the f88edded68b36aac1ef21c76d179feb47a8e24ad  (that I have locally).

**Comment by lasote** [comment]
Thanks, so timestamps seem ok.
I cannot reproduce it locally, would you have the possibility to use a conan client from a branch I can prepare? You just need to checkout the branch, create a clean python virtual environment and run `pip install -e .`
I can add some traces in the source code to see what is going on.

**Comment by lasote** [comment]
In case you could do it, this is the repo and branch:

```
git clone git@github.com:lasote/conan.git && cd conan
git checkout debug/updates_not_working
```

Just repeat the install and paste here the output.

**Comment by birsoyo** [comment]
I am having exact the same problem. 
server : JFrog Artifactory 7.33.12 rev 73312900 
macOS: 11.6.2
python: 3.8.9
conan : 1.47.0

I am using revision_mode = "scm". 

I think my state is same as reported above with times etc.

**Comment by AbrilRBS** [comment]
Closing as we didn't get the requested report commands back then
